.github/workflows/ci.yml

```diff
@@ -1,43 +1,34 @@
-name: Test
+name: Ruby
 
 on:
   push:
     branches: [main, master] # TODO: rename and get rid of 'master'
-  # Triggers the workflow on pull request events.
+    paths-ignore:
+      - '**/*.md'
   pull_request:
-    types: [opened, reopened, synchronize]
-  # Allows you to run this workflow manually from the Actions tab
+    branches: [main, master] # TODO: rename and get rid of 'master'
+    paths-ignore:
+      - '**/*.md'
   workflow_dispatch:
 
-# This allows a subsequently queued workflow run to interrupt previous runs.
-concurrency:
-  group: "${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}"
-  cancel-in-progress: true
-
 jobs:
   test:
-    runs-on: ubuntu-latest
     strategy:
       matrix:
-        ruby:
-          # https://www.ruby-lang.org/en/downloads/branches/
-          - "ruby-3.0"
-          - "ruby-3.1"
-          - "ruby-3.2"
-          - "ruby-3.3"
-          - "ruby-head"
-          # https://github.com/jruby/jruby/discussions/7915
-          - "jruby-9.4"
-          - "jruby-head"
-    name: "Test (Ruby ${{ matrix.ruby }})"
-    env:
-      SKIP_ADAPTERS: jr_jackson,nsjsonserialization,json_pure
+        ruby-version: ['3.0', '3.1', '3.2', 'jruby']
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.platform }}
+
     steps:
-      - uses: actions/checkout@v4
-      - name: Set up Ruby
-        uses: ruby/setup-ruby@v1
-        with:
-          ruby-version: ${{ matrix.ruby }}
-          bundler-cache: true
-      - run: bundle install
-      - run: bundle exec rake
+    - uses: actions/checkout@v4
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby-version }}
+        bundler-cache: false # Do not bundle install yet, need recent versions for Windows
+    - name: Run tests
+      run: |
+        gem install bundler
+        bundle install
+        bundle exec rake
+
```

Comment:
I had to refactor this a bit because older versions of Ruby default to a version of Bundler that doesn't understand the "windows" platform for the Gemfile.